truss/cli.py

```diff
@@ -175,8 +181,46 @@ def watch(
 
     if venv_dir:
         venv_dir = Path(venv_dir)
+    image_builder = ServingImageBuilderContext.run(Path(target_directory))
+    server_loader = LocalServerLoader(Path(target_directory), image_builder, port=port)
+    server_loader.watch(build_dir, venv_dir)
+
+
+@cli_group.command()
+@click.argument("target_directory", required=False)
+@click.option("--api-key", type=str, required=False, help="Your API key")
+@error_handling
+def watch(
+    target_directory: str,
+    api_key: str,
+) -> None:
+    """
+    Watches local truss directory for changes and sends patch requests to remote development truss
+
+    TARGET_DIRECTORY: A Truss directory. If none, use current directory.
+    """
     tr = _get_truss_from_directory(target_directory=target_directory)
-    tr.run_local_server_with_reload(build_dir=build_dir, work_dir=venv_dir, port=port)
+    model_name = tr.spec.config.model_name
+    if not model_name:
+        raise ValueError("'NoneType' model_name value provided in config.yaml")
+
+    # todo(@abu): refactor to BasetenRemote.AuthService
+    if not api_key:
+        api_key = os.environ.get("BASETEN_API_KEY", None)  # type: ignore
+        if not api_key:
+            raise ValueError(
+                "API key must be provided either as a flag or an environment variable BASETEN_API_KEY"
+            )
+    remote = BasetenRemote("https://app.baseten.co", api_key)
```

Comment:
Does this make sense to refactor from here to the bottom of this function to something like `BasetenRemote(...).sync_truss_to_dev_version_by_name(model_name)`?

Reply:
will be refactored once trussrc work is in